core/src/mender-client.c

```diff
@@ -1080,11 +1080,27 @@ mender_client_update_work_function(void) {
     /* reset the currently used update module */
     mender_update_module = NULL;
 
+    {
+        cJSON *deployment_id_j = cJSON_GetObjectItem(mender_client_deployment_data, "id");
+        if (NULL != deployment_id_j) {
+            deployment_id = deployment_id_j->valuestring;
```

Comment:
`cJSON_GetStringValue()` will also check that the type is actually string. However, it does provides some overhead. And you are just moving code around. So feel free to dismiss ;)